src/DotVVM.Framework/Resources/Scripts/validation/validation.ts

```diff
@@ -44,26 +44,31 @@ const createValidationHandler = (path: string) => ({
             options.validationTargetPath = path;
             // resolve target
             const context = ko.contextFor(options.sender);
-            const validationTarget = evaluator.evaluateOnViewModel(context, path);
+            const validationTarget = evaluator.evaluateExpression(context, path);
 
-            watchAndTriggerValidationErrorChanged(options, () => {
-                detachAllErrors();
-                validateViewModel(validationTarget);
-            });
+            watchAndTriggerValidationErrorChanged(options,
+                () => {
+                    detachAllErrors();
+                    validateViewModel(validationTarget);
+                });
 
             if (allErrors.length > 0) {
-                logError("validation", "Validation failed: postback aborted; errors: ", allErrors);
-                return Promise.reject(new DotvvmPostbackError({ type: "handler", handlerName: "validation", message: "Validation failed" }))
+                console.log("Validation failed: postback aborted; errors: ", allErrors);
+                return Promise.reject(new DotvvmPostbackError({
+                    type: "handler",
+                    handlerName: "validation",
+                    message: "Validation failed"
+                }))
             }
         }
-        return callback()
+        return callback();
     }
-})
+});
 
 export function init() {
     postbackHandlers["validate"] = (opt) => createValidationHandler(opt.path);
-    postbackHandlers["validate-root"] = () => createValidationHandler("dotvvm.viewModelObservables['root']");
-    postbackHandlers["validate-this"] = () => createValidationHandler("$data");
+    postbackHandlers["validate-root"] = () => createValidationHandler("/");
+    postbackHandlers["validate-this"] = () => createValidationHandler("$data"); //TODO
```

Comment:
In many cases we don't know what the "working directory" is. Because of this I don't want to support anything like `"."` / `".."` .

Regarding the empty string instead of `"/"`. This change would decrease readability in my opinion.

Reply:
We decided to continue with `'/'`. However, the `'$data'` will be removed if possible